src/api/mediators.coffee

```diff
@@ -98,4 +98,85 @@ exports.removeMediator = (urn) ->
     this.body = "Mediator with urn #{urn} has been successfully removed by #{this.authenticated.email}"
     logger.info "Mediator with urn #{urn} has been successfully removed by #{this.authenticated.email}"
   catch err
-    utils.logAndSetResponse this, 500, "Could not remove Mediator by urn  {urn} via the API: #{err}", 'error'
+    utils.logAndSetResponse this, 500, "Could not remove Mediator by urn #{urn} via the API: #{err}", 'error'
+
+exports.heartbeat = (urn) ->
+  # Must be admin
+  if not authorisation.inGroup 'admin', this.authenticated
+    utils.logAndSetResponse this, 403, "User #{this.authenticated.email} is not an admin, API access to removeMediator denied.", 'info'
+    return
+
+  urn = unescape urn
+
+  try
+    mediator = yield Mediator.findOne({ urn: urn }).exec()
+
+    if not mediator?
+      this.status = 404
+      return
+
+    heartbeat = this.request.body
```

Comment:
There isn't much to validate but I can add a check that `.uptime` always exists.